docker-compose.yml

```diff
@@ -1,44 +1,99 @@
-version: '2.1'
+version: '3.4'
+
+x-system_environment:
+  &system_environment
+  links:
+    - apisonator
+    - mailhog
+    - memcached
+    - mysql
+    - redis
+  environment:
+    - DATABASE_URL=mysql2://root@mysql/3scale_system_development
+    - REDIS_URL=redis://redis/1
+    - MEMCACHE_SERVERS=memcached:11211
+    - BACKEND_HOST=apisonator:3001
+    - SECRET_KEY_BASE=secret
+    - THREESCALE_CORE_INTERNAL_API=http://apisonator:3001/internal/
+    - CONFIG_INTERNAL_API_USER=system_app
+    - CONFIG_INTERNAL_API_PASSWORD=password
+    - SMTP_ADDRESS=mailhog
+    - SMTP_PORT=1025
+    - UNICORN_WORKERS=2
+    - FORCE_SSL=false
+    - RAILS_ENV=development
+
 services:
-  keycloak:
-    image: jboss/keycloak:3.0.0.Final
-    environment:
-      KEYCLOAK_USER: keycloak
-      KEYCLOAK_PASSWORD: development
-      KEYCLOAK_LOGLEVEL: DEBUG
+  system:
+    <<: *system_environment
+    build: .
+    command: bash -c "rm -f tmp/pids/server.pid && UNICORN_WORKERS=2 bin/rails server -b 0.0.0.0"
+    container_name: system_compose
+    depends_on:
+      - apisonator
+      - mailhog
+      - memcached
+      - redis
+      - mysql
+    ports:
+      - "3000:3000"
+
+  sidekiq:
+    <<: *system_environment
+    image: porta_system
+    command: bash -c "rake sidekiq"
+    container_name: sidekiq_compose
+    depends_on:
+      - system
+
+  mysql:
+    image: mysql:5.7
+    container_name: mysql_compose
     ports:
-      - "8080"
-      - "9990"
-  backend:
-    image: quay.io/3scale/amp:backend-2.1.0-ER1
+      - "3306:3306"
     environment:
-      RACK_ENV: preview
-      CONFIG_REDIS_PROXY: redis:6379
-      CONFIG_QUEUES_MASTER_NAME: redis:6379
-      CONFIG_WORKERS_LOG_FILE: /dev/stdout
-  backend-listener:
-    extends: backend
-    command: rackup --host 0.0.0.0 --port 3000
-    container_name: backend
+      - MYSQL_ALLOW_EMPTY_PASSWORD=true
+    healthcheck:
+        test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
+        timeout: 20s
+        retries: 10
+    volumes:
+      - mysql-data:/var/lib/mysql
+
+  memcached:
+    image: memcached
+    container_name: memcached_compose
     ports:
-      - "3001:3000"
-    depends_on:
-      - redis
-      - backend-worker
-  backend-worker:
-    extends: backend
-    command: 3scale_backend_worker run
+      - "11211:11211"
+
+  apisonator:
+    image: quay.io/3scale/apisonator:latest
+    container_name: apisonator_compose
+    ports:
+      - "3001:3001"
+    command: "3scale_backend start -p 3001"
     depends_on:
       - redis
+    environment:
+      - CONFIG_REDIS_PROXY=redis:6379/2
+      - CONFIG_QUEUES_MASTER_NAME=redis:6379/2
```

Comment:
This will make apisonator to share the same db as porta message bus.

In `system_environment` you can also specify `MESSAGE_BUS_REDIS_URL`. If missing, porta will assume `REDIS_URL`'s db +1 for the message bus (used by proxy deployments and communication with zync).

https://github.com/3scale/porta/blob/5efe0aed74c429f17575958f86d842b09e8a0a11/app/lib/three_scale/message_bus_config.rb#L44

To fix this you have 3 options:
1. Set a different redis db for apisonator, other than 1 or 2 (both taken by porta already).
2. Set a specific redis db to porta message bus other than 1 or 2 (respectively taken by porta for sidekiq and by apisonator). Do it for example with `MESSAGE_BUS_REDIS_URL=redis://redis:6379/8` in `system_environment`.
3. Set in `system_environment` both `REDIS_NAMESPACE=sidekiq` and `MESSAGE_BUS_REDIS_NAMESPACE=mbus` so sidekiq and message bus can both share the same db without risk of key collision. (The values can actually be whatever you want. These are just examples.)